rules/awsrules/models/aws_s3_bucket_invalid_acl.go

```diff
@@ -1,35 +1,32 @@
-package awsrules
+// This file generated by `tools/model-rule-gen/main.go`. DO NOT EDIT
+
+package models
 
 import (
-	"fmt"
 	"log"
 
 	"github.com/hashicorp/hcl2/hcl"
 	"github.com/wata727/tflint/issue"
 	"github.com/wata727/tflint/tflint"
 )
 
-// AwsS3BucketInvalidACLRule checks whether "aws_s3_bucket" has invalid ACL setting.
+// AwsS3BucketInvalidACLRule checks the pattern is valid
 type AwsS3BucketInvalidACLRule struct {
 	resourceType  string
 	attributeName string
-	aclTypes      map[string]bool
+	enum          []string
 }
 
 // NewAwsS3BucketInvalidACLRule returns new rule with default attributes
 func NewAwsS3BucketInvalidACLRule() *AwsS3BucketInvalidACLRule {
 	return &AwsS3BucketInvalidACLRule{
 		resourceType:  "aws_s3_bucket",
 		attributeName: "acl",
-		aclTypes: map[string]bool{
-			"private":                   true,
-			"public-read":               true,
-			"public-read-write":         true,
-			"aws-exec-read":             true,
-			"authenticated-read":        true,
-			"bucket-owner-read":         true,
-			"bucket-owner-full-control": true,
-			"log-delivery-write":        true,
```

Comment:
why was this one removed though? It seems perfectly valid to me?

Reply:
In fact, all of the above are valid according to the same link you've referenced in the PR description (https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl), so I don't quite understand the change?

Reply:
Umm, what you are saying seems right. It's weird...
The list is generated from [aws-sdk API definition](https://github.com/aws/aws-sdk-go/blob/v1.20.4/models/apis/s3/2006-03-01/api-2.json#L1071-L1079), so It may be something wrong with how to use. I will investigate that.

Reply:
Looks like that block is way out of date - hasn't been updated in 5 years...

Reply:
There were commits to add new Canned ACLs since then (https://github.com/aws/aws-sdk-go/commit/0922c232017dc730e908d7528741ceed9d30c796 for example), but that particular block was never updated :(

Reply:
https://github.com/aws/aws-sdk-go/issues/2683